src/library_dylink.js

```diff
@@ -44,20 +47,28 @@ var LibraryDylink = {
   },
 
   $GOT: {},
+  $CurrentModuleWeakSymbols: '=new Set({{{ JSON.stringify(Array.from(WEAK_IMPORTS)) }}})',
 
   // Create globals to each imported symbol.  These are all initialized to zero
   // and get assigned later in `updateGOT`
   $GOTHandler__internal: true,
-  $GOTHandler__deps: ['$GOT'],
+  $GOTHandler__deps: ['$GOT', '$CurrentModuleWeakSymbols'],
   $GOTHandler: {
     'get': function(obj, symName) {
-      if (!GOT[symName]) {
-        GOT[symName] = new WebAssembly.Global({'value': '{{{ POINTER_WASM_TYPE }}}', 'mutable': true});
+      var rtn = GOT[symName];
+      if (!rtn) {
+        rtn = GOT[symName] = new WebAssembly.Global({'value': '{{{ POINTER_WASM_TYPE }}}', 'mutable': true});
 #if DYLINK_DEBUG
         err("new GOT entry: " + symName);
 #endif
       }
-      return GOT[symName]
+      if (!CurrentModuleWeakSymbols.has(symName)) {
+        // This is a non-weak (i.e. strong) reference to the symbol so we take
+        // it as such.  Any undefined symbols that are strongly referenced are 
+        // then reported in `reportUndefinedSymbols`.
+        rtn.strong = true;
```

Comment:
Why check "if !weak" and then mark "strong"? It seems like it would be slightly simpler to mark the weak ones instead.

Which is more common? If weak symbols are rarer, that could be another reason to mark them and not strong ones.

Reply:
The reason is that any number of modules could contain an undefined reference to this symbol.   If any one of them is strong we need to report the error.

This way to track "is there any strong reference to this symbol" rather than "is there any weak reference to this symbol".   The former tells is that we need to report that symbol if its undefined.  That later, on its own doesn't tell us enough, because we need to know if *all* the references are weak in order to suppress the undefined symbol warning.

Reply:
Is it not symmetric, though? I mean, this line could be `rtn.weak = true` (and the condition above flipped). Then the single place that uses this below could replace `!GOT[symName].strong` with `GOT[symName].weak`. That seems slightly simpler to me, and more efficient assuming weak symbols are rarer. But maybe I'm missing something here.